pubsub/pubsub.go

```diff
@@ -22,16 +22,30 @@ import (
 
 // PubSub is the interface for message buses.
 type PubSub interface {
+	BulkMessager
 	Init(metadata Metadata) error
 	Features() []Feature
 	Publish(req *PublishRequest) error
 	Subscribe(ctx context.Context, req SubscribeRequest, handler Handler) error
 	Close() error
 }
 
+// BulkMessager is the interface defining BulkPublish and BulkSubscribe definitions for message buses
+type BulkMessager interface {
+	BulkPublish(req *BulkPublishRequest) (BulkPublishResponse, error)
+	BulkSubscribe(ctx context.Context, req SubscribeRequest, handler BulkHandler) (BulkSubscribeResponse, error)
```

Comment:
Return type be error only - BulkHandler to indicate correct status for contained items in bulk.